emily/handler/tests/integration/deposit.rs

```diff
@@ -532,6 +548,127 @@ async fn get_deposits_for_recipient() {
+                DEPOSIT_LOCK_TIME,
+                DEPOSIT_MAX_FEE,
+                &amounts,
+                &reclaim_user_script,
+            );
+
+            for bitcoin_tx_output_index in 0..*num_deposits {
+                let tx_output_index = bitcoin_tx_output_index as usize;
+                let recipient = recipients[tx_output_index].clone();
+                let reclaim_script = reclaim_scripts[tx_output_index].clone();
+                let deposit_script = deposit_scripts[tx_output_index].clone();
+                // Make the create request.
+                let request = CreateDepositRequestBody {
+                    bitcoin_tx_output_index,
+                    bitcoin_txid: bitcoin_txid.clone(),
+                    deposit_script: deposit_script.clone(),
+                    reclaim_script: reclaim_script.clone(),
+                    transaction_hex: transaction_hex.clone(),
+                };
+                create_requests.push(request);
+                // Store the expected deposit info that should come from it.
+                let expected_deposit_info = DepositInfo {
+                    amount: DEPOSIT_AMOUNT_SATS,
+                    bitcoin_tx_output_index,
+                    bitcoin_txid: bitcoin_txid.clone(),
+                    last_update_block_hash: BLOCK_HASH.into(),
+                    last_update_height: BLOCK_HEIGHT,
+                    recipient: recipient.clone(),
+                    status: testing_emily_client::models::Status::Pending,
+                    reclaim_script: reclaim_script,
+                    deposit_script: deposit_script,
+                };
+                expected_deposit_infos.push(expected_deposit_info);
+            }
+        }
+        // Add the pubkey data to the pubkey data hashmap that stores what
+        // we expect to see from the pubkey.
+        expected_pubkey_data.insert(hex::encode(pubkey), expected_deposit_infos.clone());
+    }
+
+    // The size of the chunks to grab from the api.
+    let chunksize = 2;
+
+    // Act.
+    // ----
+    batch_create_deposits(&configuration, create_requests).await;
+
+    let mut actual_pubkey_data: HashMap<String, Vec<DepositInfo>> = HashMap::new();
+    for pubkey in expected_pubkey_data.keys() {
+        // Loop over the api calls to get all the deposits for the pubkey.
+        let mut gotten_deposit_info_chunks: Vec<Vec<DepositInfo>> = Vec::new();
+        let mut next_token: Option<Option<String>> = None;
+        loop {
+            let response = apis::deposit_api::get_deposits_for_reclaim_pubkeys(
+                &configuration,
+                pubkey,
+                next_token.as_ref().and_then(|o| o.as_deref()),
+                Some(chunksize),
+            )
+            .await
+            .expect("Received an error after making a valid get deposits for pubkey api call.");
+            gotten_deposit_info_chunks.push(response.deposits);
+            next_token = response.next_token;
```

Comment:
nit: `response.next_token.flatten()` may simplify the `next_token.as_ref().and_then` thingy